go/cmd/dolt/dolt.go

```diff
@@ -154,6 +155,14 @@ func runMain() int {
 	warnIfMaxFilesTooLow()
 
 	ctx := context.Background()
+
+	err := nbs.MaybeMigrateFileManifest(ctx, dbfactory.DoltDataDir)
```

Comment:
As mentioned in chat, we can do this optimistically (i.e., anytime a new dolt client runs against a directory.) But we don't _have_ to do this until the first use of GC version is needed (first GC run I guess...). On the balance, if we expect GC to be commonly used, the tradeoff doesn't matter too much. If we expect GC to remain experimental for a while and we want to ease into the impacts of introducing it, we might want to take the other approach...